### DISCUSSION TITLE
Failed test doesn't show line number in my code

### DISCUSSION THREAD

**Discussion by benjamin-inspectly** [leancodepl/patrol#1615] [Q&A] [answered]
Hi Patrol and all others,

Thank you for always taking your time to answer my newbie questions :-)

I was wondering if my error messages on my tests are normal or not.
I always get the assertion error but I never get what line number my test failed at.

In below case, my test failed in a "pumpAndSettle" step, but if I have multiple of those on my test, I'll have to run the test locally and hope for the same error locally as in my CI environment.

Did I miss something during setup?

Error message:

java.lang.AssertionError: Dart test failed: critical_workflows.photoDoc.0003_Create_full_2_of_each_doc_test

══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
The following assertion was thrown running a test:
pumpAndSettle timed out

When the exception was thrown, this was the stack:
#0      WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:707:11)
<asynchronous suspension>
#1      TestAsyncUtils.guard.<anonymous closure> (package:flutter_test/src/test_async_utils.dart:114:7)
<asynchronous suspension>

The test description was:
PhotoDoc_0002 - Create full photodoc with 2 of each doc items
═════════════════════════════════════════════════════════════════

at pl.leancode.patrol.PatrolJUnitRunner.runDartTest(PatrolJUnitRunner.java:130)
at com.ikontrol.app_new.MainActivityTest.runDartTest(MainActivityTest.java:29)

**Comment by benjamin-inspectly** [comment]
@bartekpacia  sorry to tag you directly in this old message you have nothing to do with. You answered some of my other questions, so maybe I'm lucky you can help me with this as well? :)
This is quite the problem for me it turns out, since some of my tests are flaky and I jest get the pumpAndSettle timed out.. When I run my test, most often all works, but 1/4 of the times I get an error, and I would love to make it more robust. It's just difficult without the line-number :/

**Reply by cambiph** [reply]
Did you check the build-directory? Android generates a HTML report with more info under `build > app > reports > androidTests > connected > index.html`

**Reply by bartekpacia** [reply]
Thanks @cambiph for helping out – that's what I suggest doing as well:)

Another not perfect but working solution is `adb logcat | grep 'flutter'` to see Dart stacktrace, `print()`s, etc.

**Reply by benjamin-inspectly** [reply]
Thanks for your reply @cambiph but in the original description, the content is actually copied from that page and looks like this:

![image](https://github.com/leancodepl/patrol/assets/114057836/18fb6a1c-451e-42d3-b20a-18bb94afd3ca)

It doesn't say anything about the linenumber because it is always the PatrolJUnitRunner and my own MainActivityTest class that generates the error.

Isn't this a problem for others besides me or are there normally linenumbers in your end that references the patrol test-case?

**Reply by bartekpacia** [reply, marked as answer]
I think you're hitting this Dart bug: https://github.com/dart-lang/sdk/issues/46318

There's no good workaround, best you can do is sprinkle some `print()`s and see in `adb logcat` where it gets stuck.

**Reply by benjamin-inspectly** [reply]
Hm okay, sorry to hear that :) Thanks for your reply tho :)

I'll give it a shot :)

**Reply by cambiph** [reply]
Do note that `print()` statements are only printed to the console when using `patrol develop` not while using `patrol test`. I saw an issue for this in the issue list.
